packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts

```diff
@@ -194,6 +194,13 @@ ruleTester.run('prefer-optional-chain', rule, {
     'match && match$1 !== undefined;',
     'foo !== null && foo !== undefined;',
     "x['y'] !== undefined && x['y'] !== null;",
+    // private properties
+    'this.#a && this.#b;',
+    '!this.#a || !this.#b;',
+    'a.#foo && a.#foo.bar;',
+    '!a.#foo || !a.#foo.bar;',
+    'a.#foo?.bar;',
+    '!a.#foo?.bar;',
```

Comment:
More test cases to consider:

```js
foo().#a
a.b.#a
new A().#b
```

I have no idea why this rule is so strict about what expressions it can lint, but you can in fact use private properties on all expressions, just that some may have nonsensical behaviors (e.g. `(typeof a).#a`). It shouldn't crash the linter anyway.

Reply:
Can I fix the rule to not intentionally throw? this seems irrational to me.
It's better to not show a suggestion than to just break and do nothing

Reply:
I would love to change this rule to not intentionally throw, or at least catch the error.
The rule logic now has many combinations that could be missed and better be gracefully ignored.

@Josh-Cena @JoshuaKGoldberg @bradzacher WDYT?

Reply:
Yeah as Brad said in https://github.com/typescript-eslint/typescript-eslint/pull/6028#discussion_r1026051986, I think the rule should no longer throw. Throwing is useful for us finding missing combinations but is inconvenient for users.

Reply:
Oh, so I misunderstood his comment

Reply:
I don't know if throwing is specifically a bad thing - it's a good way to explicitly tell the user "hey the rule hasn't worked right here", rather than just failing silently and probably doing a false-negative.

The big problem is that there are a lot of unhandled cases now due to the previous refactor. As I mentioned in the comment Josh linked:

> Previously the call to getText was always guarded behind isValidChainTarget which (I believe) would have enforced the expected ALLOWED_NON_COMPUTED_PROP_TYPES constraint.
> ...
> Considering the code now has an unguarded call to getText, this switch/case needs to be updated to handle all possible cases!